src/Generation/Generator/Renderer/Public/ReturnType/From/RenderableArrayReturnType.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace Generator.Renderer.Public;
+
+internal abstract class RenderableArrayReturnType<T> : IRenderableReturnType where T : GirModel.Type
```

Comment:
Please remove additional base class. Implement the `Supports` method in each class individually. It makes each concrete implementation easier to understand and to change.